packages/@sanity/default-layout/src/navbar/search/components/SearchHeader.tsx

```diff
@@ -0,0 +1,136 @@
+      filtersVisible,
+      result: {loading},
+      terms,
+    },
+  } = useSearchState()
+
+  const handleFiltersToggle = useCallback(() => dispatch({type: 'FILTERS_TOGGLE'}), [dispatch])
+  const handleQueryChange = useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) =>
+      dispatch({type: 'TERMS_QUERY_SET', query: e.currentTarget.value}),
+    [dispatch]
+  )
+  const handleQueryClear = useCallback(() => {
+    dispatch({type: 'TERMS_QUERY_SET', query: ''})
+  }, [dispatch])
+
+  // Focus filter button (when filters are hidden after initial mount)
+  useEffect(() => {
+    if (isMountedRef?.current && !filtersVisible) {
+      filterButtonElement?.focus()
+    }
+  }, [filterButtonElement, filtersVisible])
+
+  useEffect(() => {
+    isMountedRef.current = true
+  }, [])
+
+  return (
+    <SearchHeaderWrapper>
+      <Flex align="center" flex={1}>
+        {/* Search field */}
+        <Box
+          flex={1}
+          paddingLeft={onClose ? 2 : 1}
+          paddingRight={onClose ? 2 : 0}
+          paddingY={onClose ? 2 : 1}
+        >
+          <CustomTextInput
+            autoComplete="off"
+            border={false}
+            clearButton={!!terms.query}
+            fontSize={2}
+            icon={loading ? <AlignedSpinner /> : SearchIcon}
+            onChange={handleQueryChange}
+            onClear={handleQueryClear}
+            placeholder="Search"
+            ref={setHeaderInputRef}
+            smallClearButton
+            spellCheck={false}
+            value={terms.query}
+          />
+        </Box>
+
+        {/* Filter toggle */}
+        <Card borderLeft={!!onClose} padding={onClose ? 2 : 1}>
+          <FilterButton>
+            <Button
+              aria-controls="search-type-filters"
+              aria-expanded={filtersVisible}
+              aria-label="Filter"
+              height="fill"
+              icon={ControlsIcon}
+              mode="bleed"
+              onClick={handleFiltersToggle}
+              padding={3}
+              ref={setFilterButtonRef}
+              selected={filtersVisible}
+              tone="default"
+            />
+            {terms.types.length > 0 && <NotificationBadge>{terms.types.length}</NotificationBadge>}
+          </FilterButton>
+        </Card>
+
+        {/* (Fullscreen) Close button */}
+        {onClose && (
+          <Card borderLeft padding={2}>
+            <Button aria-label="Close search" icon={CloseIcon} mode="bleed" onClick={onClose} />
+          </Card>
+        )}
+      </Flex>
+    </SearchHeaderWrapper>
+  )
+}
+
+const AlignedSpinner = styled(Spinner)`
+  svg {
+    width: 20px;
+    vertical-align: bottom !important;
+  }
+`
+
+const FilterButton = styled(Box)`
+  position: relative;
+`
+
+const SearchHeaderWrapper = styled(Box)`
+  border-bottom: 1px solid ${hues.gray[200].hex};
+  flex-shrink: 0;
+`
+
+const NotificationBadge = styled.div`
+  align-items: center;
+  background: ${hues.blue[700].hex};
+  color: ${white.hex};
```

Comment:
Are we sure we dont want to use colors from theme for this instead? Same goes for hues? I think using sanity color directly is discouraged. Will have to be rewritten for dark theme in v3 I think.

Reply:
You're definitely right here! Dropped the `@sanity/colors` dependency throughout and now pull from the theme.